cfme/v2v/migrations.py

```diff
@@ -682,13 +682,28 @@ def create(self, name, infra_map, vm_list, description=None, csv_import=False,
 
 
 @navigator.register(InfrastructureMappingCollection, 'All')
+class AllMappings(CFMENavigateStep):
+    VIEW = MigrationDashboardView
+    prerequisite = NavigateToAttribute('appliance.server', 'LoggedIn')
+
+    def step(self):
+        if self.obj.appliance.version < '5.10':
+            self.prerequisite_view.navigation.select('Compute', 'Migration')
+        else:
+            self.prerequisite_view.navigation.select(
+                'Compute', 'Migration', 'Infrastructure Mappings')
+
+
 @navigator.register(MigrationPlanCollection, 'All')
 class All(CFMENavigateStep):
     VIEW = MigrationDashboardView
     prerequisite = NavigateToAttribute('appliance.server', 'LoggedIn')
 
     def step(self):
-        self.prerequisite_view.navigation.select('Compute', 'Migration')
+        if self.obj.appliance.version < '5.10':
+            self.prerequisite_view.navigation.select('Compute', 'Migration')
+        else:
+            self.prerequisite_view.navigation.select('Compute', 'Migration', 'Overview')
```

Comment:
Consider just checking the type of `self.obj` in order to select `'Overview'` or `'Infrastructure Mappings'` for navigation. Then this can be a single nav class registered against both `InfrastructureMappingCollection` and `MigrationPlanCollection`

Reply:
I need to create 2 different views for Navigation to `All` for `InfraMapping` and `MigrationPlan` so wouldn't it be easier just to split them in two classes?